### PR TITLE
Updates to debugDisableShadows

### PR DESCRIPTION
**Pull request by tvolkert**
1. Make BoxShadow.toPaint() add a stroke so that we can see where
   shadows would have otherwise been painted.
2. Make CupertinoSwitch use BoxShadow.toPaint() so that it respects
   the `debugDisableShadows` flag.

### CODE REVIEW

**Comment by tvolkert** [comment]
This change produces a weird difference in `shadow.BoxDecoration.disabled.png`:

[Before](https://github.com/flutter/goldens/blob/a3923c1a6ed192ad55f18bef6999422d11fd7b1b/packages/flutter/test/widgets/shadow.BoxDecoration.disabled.png):
before

[After](https://github.com/flutter/goldens/blob/7b308e37d33605bbb535df2b5d596a8697ac78bd/packages/flutter/test/widgets/shadow.BoxDecoration.disabled.png)
after

File: packages/flutter/lib/src/cupertino/thumb_painter.dart [hunk 1]
```diff
@@ -35,17 +33,17 @@ class CupertinoThumbPainter {
...
-    paint
-      ..color = color
-      ..maskFilter = null;
-    canvas.drawRRect(rrect, paint);
+    final RRect rrect = new RRect.fromRectAndRadius(
+      rect,
+      new Radius.circular(rect.shortestSide / 2.0),
+    );
+
+    final Paint shadowPaint = new BoxShadow(
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/thumb_painter.dart]: let's create that in the constructor

Comment (Hixie) [reply packages/flutter/lib/src/cupertino/thumb_painter.dart]: this still applies :-)

File: packages/flutter/lib/src/painting/box_shadow.dart [hunk 2]
```diff
@@ -71,8 +71,10 @@ class BoxShadow {
       ..color = color
       ..maskFilter = new MaskFilter.blur(BlurStyle.normal, blurSigma);
     assert(() {
-      if (debugDisableShadows)
-        result.maskFilter = null;
+      if (debugDisableShadows) {
+        result
+          ..maskFilter = null;
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/box_shadow.dart]: i assume you meant to revert this :-)

File: packages/flutter/lib/src/cupertino/thumb_painter.dart (hunk 1 above)
Comment (tvolkert) [reply packages/flutter/lib/src/cupertino/thumb_painter.dart]: Weird - hadn't seen this before.  Done.

File: packages/flutter/lib/src/painting/box_shadow.dart (hunk 2 above)
Comment (tvolkert) [reply packages/flutter/lib/src/painting/box_shadow.dart]: Yep - done

File: packages/flutter/lib/src/cupertino/thumb_painter.dart [hunk 3]
```diff
@@ -16,14 +14,22 @@ class CupertinoThumbPainter {
...
+  }
 
   /// The color of the interior of the thumb.
   final Color color;
 
   /// The color of the shadow case by the thumb.
   final Color shadowColor;
 
+  /// The paint used to draw the shadow case by the thumb.
+  Paint _shadowPaint;
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/thumb_painter.dart]: this can be final if you set it in the constructor initialiser list

**Comment by Hixie** [comment]
LGTM

**Comment by tvolkert** [comment]
Merging on red because the redness is a suspected flake that we're unable to restart.

**Pull request merged**